### PR TITLE
Use IAsyncEnumerable for GetRules

### PR DESCRIPTION
**Pull request by JoshLove-msft**

### CODE REVIEW

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpRuleManager.cs [hunk 1]
```diff
@@ -225,34 +225,32 @@ static AmqpRuleManager()
...
-        /// Get all rules associated with the subscription.
+        /// Get rules associated with the subscription.
         /// </summary>
-        ///
+        /// <param name="skip">The number of rules to skip when retrieving the next set of rules.</param>
         /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
-        ///
         /// <returns>Returns a list of rules description</returns>
-        public override async Task<List<RuleProperties>> GetRulesAsync(CancellationToken cancellationToken) =>
+        public override async Task<List<RuleProperties>> GetRulesAsync(int skip, CancellationToken cancellationToken) =>
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpRuleManager.cs:233]: Does this ever some in at anything but 0?

**Review by jsquire — APPROVED** [review]

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpRuleManager.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpRuleManager.cs:233]: It should be updated between calls in the IAsyncEnumerable method.

Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpRuleManager.cs:233]: See **https://github.com/Azure/azure-sdk-for-net/pull/29130/files#diff-fbb54ffd1afffda0126f0de0ef4313b6aa4062f9b69194a3cdeb7d6d668e78bbR211**

Comment (jsquire) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpRuleManager.cs:233]: Gotcha;  I missed that the rule manager was using it!

**Pull request merged**